x/supply/internal/keeper/keeper.go

```diff
@@ -20,32 +20,26 @@ type Keeper struct {
 }
 
 type permAddr struct {
```

Comment:
maybe we should move these perm-related types and funcs to `/types/permissions.go`

Reply:
Agreed, should I add a Add/Remove Permission functions since the permissions field will be private?